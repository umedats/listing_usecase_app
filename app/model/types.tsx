import { Subscription as SDKSubscription } from "@/app/sdk/notifications";

/**
 * The Issue returned from the Listing API.
 */
export interface Issue {
  code: string;
  message: string;
  severity: string;
  attributeNames?: string[];
}

/**
 * The result of the Listings Items API submission.
 */
export interface ListingSubmissionResult {
  sku: string;
  status: string;
  submissionId: string;
  issues?: Issue[];
}

/**
 * Relevant information from Listings GET API call.
 */
export interface Listing {
  attributes: object;
  issues?: Issue[];
  productType?: string;
}

/**
 * A type which represents a patch for the contribution.
 */
export interface Patch {
  op: string;
  path: string;
  value: any;
}

/**
 * A type for the alert dialog state.
 */
export interface AlertDialogState {
  showAlert: boolean;
  alertTitle: string;
  alertContent: string;
}

/**
 * A type which captures the SP API request and response.
 */
export interface SPAPIRequestResponse {
  apiName: string;
  apiDocumentationLink: string;
  request: object;
  response: object;
}

/**
 * A type which capture the information about a feed.
 */
export interface Feed {
  feedId: string;
  feedType: string;
  createdTime: string;
  processingStatus: string;
  resultFeedDocumentId: string;
}

/**
 * A type which capture the information about a Subscription.
 */
export interface Subscription {
  notificationType: string;
  eventBusName?: string;
  eventBusFilterRule?: string;
  sqsTargetId?: string;
  sqsQueueName?: string;
  subscriptionId: string;
}

/**
 * A type which captures the raw SDK subscription along with the
 * notification type.
 */
export interface RawSubscription {
  notificationType: string;
  subscription: SDKSubscription;
}
