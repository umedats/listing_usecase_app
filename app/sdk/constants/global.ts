const APP_NAME = "SellingPartnerAPIListingSampleApplication";
const APP_VERSION = "1.0";
const APP_LANGUAGE = "TypeScript";
const APP_LANGUAGE_VERSION = "5.0";
/**
 * Header data to be passed to all the SP-API.
 */
export const HEADER_DATA = {
  "User-Agent": `${APP_NAME}/${APP_VERSION} (Language=${APP_LANGUAGE}/${APP_LANGUAGE_VERSION})`,
};
