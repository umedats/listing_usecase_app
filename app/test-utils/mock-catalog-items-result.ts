export const MOCK_CATALOG_ITEMS_API_RESPONSE: object = {
  numberOfResults: 1,
  items: [
    {
      asin: "B00004SPEC",
      productTypes: [
        {
          marketplaceId: "ATVPDKIKX0DER",
          productType: "KITCHEN",
        },
      ],
      summaries: [
        {
          marketplaceId: "ATVPDKIKX0DER",
          adultProduct: false,
          autographed: false,
          brand: "KRUPS",
          browseClassification: {
            displayName: "Countertop Blenders",
            classificationId: "289915",
          },
          itemClassification: "BASE_PRODUCT",
          itemName: "Krups 243-70 Power X Plus Combi Blender, DISCONTINUED",
          manufacturer: "KRUPS",
          memorabilia: false,
          modelNumber: "243-70",
          packageQuantity: 1,
          partNumber: "243-70",
          style: "Modern",
          tradeInEligible: false,
          websiteDisplayGroup: "kitchen_display_on_website",
          websiteDisplayGroupName: "Kitchen",
        },
      ],
    },
  ],
};

export const MOCK_CATALOG_ITEMS_API_EMPTY_RESPONSE: object = {
  numberOfResults: 0,
};
