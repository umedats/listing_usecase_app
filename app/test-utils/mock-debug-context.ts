import { SPAPIRequestResponse } from "@/app/model/types";

export const MOCK_SP_API_RESPONSE: SPAPIRequestResponse = {
  apiName: "GetListingsItem",
  apiDocumentationLink: "https://amazon.com",
  request: {
    marketplaceId: "ATVPDKIKX0DER",
    locale: "en_US",
    asin: "A1F83G8C2ARO7P",
  },
  response: {
    item: "Dummy Item",
  },
};
