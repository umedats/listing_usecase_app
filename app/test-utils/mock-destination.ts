import {
  CreateDestinationResponse,
  Destination,
} from "@/app/sdk/notifications";

export const MOCK_DESTINATION: Destination = {
  name: "TestDestination",
  destinationId: "5fe08e61-c5ec-4427-9f25-24a7b630cce2",
  resource: {
    eventBridge: {
      name: "sellingpartnerapi.amazon.com/amzn1.sellerapps.app.15a75829-cd4a-4efc-b947-0cc39d874577",
      region: "us-east-1",
      accountId: "123456",
    },
  },
};

export const MOCK_CREATE_DESTINATION_RESPONSE: CreateDestinationResponse = {
  payload: MOCK_DESTINATION,
  errors: {},
};
