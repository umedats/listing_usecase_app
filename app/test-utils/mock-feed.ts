export const MOCK_FEED_1 = {
  feedId: "1000",
  feedType: "JSON_LISTINGS_FEED",
  createdTime: "2023-10-05T04:23:58+0000",
  processingStatus: "IN_PROGRESS",
  resultFeedDocumentId:
    "amzn1.tortuga.3.ed4cd0d8-447b-4c22-96b5-52da8ace1207.T3YUVYPGKE9BMY",
};

export const MOCK_FEED_2 = {
  feedId: "2000",
  feedType: "JSON_LISTINGS_FEED",
  createdTime: "2023-10-05T04:23:58+0000",
  processingStatus: "IN_PROGRESS",
};

export const MOCK_CREATE_FEED_DOCUMENT_SUCCESS = {
  feedDocumentId: "123456",
  url: "https:feed.document/url",
};

export const MOCK_CREATE_FEED_SUCCESS = {
  feedId: "123456789",
};
