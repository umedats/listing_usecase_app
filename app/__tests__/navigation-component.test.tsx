import { render, screen } from "@testing-library/react";
import React from "react";
import { NextIntlProvider } from "next-intl";
import translations from "@/app/internationalization/translations/en-US.json";
import { AppRouterInstance } from "next/dist/shared/lib/app-router-context";
import NavigationComponent from "@/app/[locale]/navigation-component";
import {
  SettableSettingsState,
  SettingsContext,
  SettingsState,
} from "@/app/context/settings-context-provider";
import {
  INCOMPLETE_SETTINGS,
  MOCK_SETTINGS,
  MOCK_VENDOR_CODE_SETTINGS,
} from "@/app/test-utils/mock-settings";
import userEvent from "@testing-library/user-event";
import DebuggingContextProvider, {
  DebugState,
  SettableDebugState,
} from "@/app/context/debug-context-provider";
import { SPAPIRequestResponse } from "@/app/model/types";
import { MOCK_SP_API_RESPONSE } from "@/app/test-utils/mock-debug-context";
import { SETTINGS_PAGE_PATH, US_LOCALE } from "@/app/constants/global";

const mockUsePathname = jest.fn();
const mockUseRouter = jest.fn();
const mockPush = jest.fn();

const mockSetSettingsState = (settingsState: SettingsState) => {};
const INCOMPLETE_SETTINGS_STATE: SettingsState = {
  settings: INCOMPLETE_SETTINGS,
  settingsExist: false,
};
const INCOMPLETE_SETTINGS_CONTEXT: SettableSettingsState = {
  settingsState: INCOMPLETE_SETTINGS_STATE,
  setSettingsState: mockSetSettingsState,
};

const COMPLETE_SETTINGS_STATE: SettingsState = {
  settings: MOCK_SETTINGS,
  settingsExist: true,
};
const COMPLETE_SETTINGS_CONTEXT: SettableSettingsState = {
  settingsState: COMPLETE_SETTINGS_STATE,
  setSettingsState: mockSetSettingsState,
};

const VENDOR_CODE_SETTINGS_STATE: SettingsState = {
  settings: MOCK_VENDOR_CODE_SETTINGS,
  settingsExist: true,
};
const VENDOR_CODE_SETTINGS_CONTEXT: SettableSettingsState = {
  settingsState: VENDOR_CODE_SETTINGS_STATE,
  setSettingsState: mockSetSettingsState,
};

const CREATE_OFFER_PATH = "/create-offer";

jest.mock("next/navigation", () => ({
  usePathname() {
    return mockUsePathname();
  },
  useRouter(): AppRouterInstance {
    return {
      back: jest.fn(),
      forward: jest.fn(),
      refresh: jest.fn(),
      push: mockPush,
      replace: jest.fn(),
      prefetch: jest.fn(),
    };
  },
}));

function renderNavigationComponent(
  settingsState: SettableSettingsState,
  debugState?: DebugState,
) {
  return render(
    <NextIntlProvider locale={US_LOCALE} messages={translations}>
      <DebuggingContextProvider initialDebugState={debugState}>
        <SettingsContext.Provider value={settingsState}>
          <NavigationComponent />
        </SettingsContext.Provider>
      </DebuggingContextProvider>
    </NextIntlProvider>,
  );
}

describe("Test for the Navigation component", () => {
  beforeEach(() => {
    mockPush.mockClear();
  });

  test("Snapshot test for Testing Navigation Component in Home Page", async () => {
    mockUsePathname.mockImplementation(() => "/");
    mockUseRouter.mockImplementation(() => ({ route: "/" }));
    const { asFragment } = renderNavigationComponent(COMPLETE_SETTINGS_CONTEXT);
    expect(asFragment()).toMatchSnapshot();
  });

  test("Snapshot test for Testing Navigation Component in Create Offer page", async () => {
    mockUsePathname.mockImplementation(() => CREATE_OFFER_PATH);
    const routeContext = new Map<string, SPAPIRequestResponse[]>([
      [CREATE_OFFER_PATH, [MOCK_SP_API_RESPONSE]],
    ]);
    const { asFragment } = renderNavigationComponent(
      COMPLETE_SETTINGS_CONTEXT,
      {
        routeContext: routeContext,
      },
    );

    expect(asFragment()).toMatchSnapshot();
    await userEvent.click(screen.getByTestId("createOffer"));
    expect(routeContext.get(CREATE_OFFER_PATH)?.length).toBeFalsy();
  });

  test("verifies if the user is navigated to home page on click of home button.", async () => {
    mockUsePathname.mockImplementation(() => CREATE_OFFER_PATH);
    renderNavigationComponent(INCOMPLETE_SETTINGS_CONTEXT);
    await userEvent.click(screen.getByTestId("homeButton"));

    expect(mockPush.mock.calls[0][0]).toStrictEqual("/");
  });

  test("verifies if the user is navigated to Settings page if the settings are incomplete", async () => {
    mockUsePathname.mockImplementation(() => CREATE_OFFER_PATH);
    renderNavigationComponent(INCOMPLETE_SETTINGS_CONTEXT);
    await userEvent.click(screen.getByTestId("createOffer"));

    expect(mockPush.mock.calls[0][0]).toStrictEqual(SETTINGS_PAGE_PATH);
  });

  test("verifies if the user is navigated to same page if the settings are complete", async () => {
    mockUsePathname.mockImplementation(() => CREATE_OFFER_PATH);
    renderNavigationComponent(COMPLETE_SETTINGS_CONTEXT);
    await userEvent.click(screen.getByTestId("createOffer"));

    expect(mockPush.mock.calls[0][0]).toStrictEqual(CREATE_OFFER_PATH);
  });

  test("verifies that the createOffer button is disabled for Vendor Codes", async () => {
    mockUsePathname.mockImplementation(() => "/");
    mockUseRouter.mockImplementation(() => ({ route: "/" }));
    const { asFragment } = renderNavigationComponent(
      VENDOR_CODE_SETTINGS_CONTEXT,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
