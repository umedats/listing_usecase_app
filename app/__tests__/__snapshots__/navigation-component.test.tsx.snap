// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the Navigation component Snapshot test for Testing Navigation Component in Create Offer page 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-enyiw4-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root css-ur2jdm-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1sgem27-MuiButtonBase-root-MuiIconButton-root"
          data-testid="homeButton"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-4vb5ob-MuiAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <span
          aria-label="Workflow to submit listing updates in bulk. See https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#creating-and-updating-listings-items-in-bulk to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="bulkListing"
            tabindex="0"
            type="button"
          >
            Bulk Listing
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          aria-label="Workflow to list an item that does not already exist in the Amazon catalog. Refer to https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#list-an-item-that-does-not-already-exist-in-the-amazon-catalog to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="createListing"
            tabindex="0"
            type="button"
          >
            Create Listing
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          aria-label="Workflow to list an offer for an item that already exists in the Amazon catalog. Refer to https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#list-an-offer-for-an-item-that-already-exists-in-the-amazon-catalog to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1khkkm6-MuiButtonBase-root-MuiButton-root"
            data-testid="createOffer"
            tabindex="0"
            type="button"
          >
            Create Offer
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          aria-label="Workflow to delete an existing listings item. See https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#deleting-a-listings-item to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="deleteListing"
            tabindex="0"
            type="button"
          >
            Delete Listing
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          aria-label="Workflow to receive notifications through Amazon EventBridge. See https://developer-docs.amazon.com/sp-api/docs/notifications-api-v1-use-case-guide#notification-workflows to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="notifications"
            tabindex="0"
            type="button"
          >
            Notifications
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          aria-label="Workflow to update an existing listings item. Refer to https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#updating-a-listings-item to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="updateListing"
            tabindex="0"
            type="button"
          >
            Update Listing
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Test for the Navigation component Snapshot test for Testing Navigation Component in Home Page 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-enyiw4-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root css-ur2jdm-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1sgem27-MuiButtonBase-root-MuiIconButton-root"
          data-testid="homeButton"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-4vb5ob-MuiAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <span
          aria-label="Workflow to submit listing updates in bulk. See https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#creating-and-updating-listings-items-in-bulk to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="bulkListing"
            tabindex="0"
            type="button"
          >
            Bulk Listing
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          aria-label="Workflow to list an item that does not already exist in the Amazon catalog. Refer to https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#list-an-item-that-does-not-already-exist-in-the-amazon-catalog to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="createListing"
            tabindex="0"
            type="button"
          >
            Create Listing
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          aria-label="Workflow to list an offer for an item that already exists in the Amazon catalog. Refer to https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#list-an-offer-for-an-item-that-already-exists-in-the-amazon-catalog to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="createOffer"
            tabindex="0"
            type="button"
          >
            Create Offer
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          aria-label="Workflow to delete an existing listings item. See https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#deleting-a-listings-item to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="deleteListing"
            tabindex="0"
            type="button"
          >
            Delete Listing
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          aria-label="Workflow to receive notifications through Amazon EventBridge. See https://developer-docs.amazon.com/sp-api/docs/notifications-api-v1-use-case-guide#notification-workflows to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="notifications"
            tabindex="0"
            type="button"
          >
            Notifications
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          aria-label="Workflow to update an existing listings item. Refer to https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#updating-a-listings-item to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="updateListing"
            tabindex="0"
            type="button"
          >
            Update Listing
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Test for the Navigation component verifies that the createOffer button is disabled for Vendor Codes 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-enyiw4-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root css-ur2jdm-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1sgem27-MuiButtonBase-root-MuiIconButton-root"
          data-testid="homeButton"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-4vb5ob-MuiAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <span
          aria-label="Workflow to submit listing updates in bulk. See https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#creating-and-updating-listings-items-in-bulk to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="bulkListing"
            tabindex="0"
            type="button"
          >
            Bulk Listing
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          aria-label="Workflow to list an item that does not already exist in the Amazon catalog. Refer to https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#list-an-item-that-does-not-already-exist-in-the-amazon-catalog to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="createListing"
            tabindex="0"
            type="button"
          >
            Create Listing
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          aria-label="Vendor Codes cannot create offers at this time"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="createOffer"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Create Offer
          </button>
        </span>
        <span
          aria-label="Workflow to delete an existing listings item. See https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#deleting-a-listings-item to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="deleteListing"
            tabindex="0"
            type="button"
          >
            Delete Listing
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          aria-label="Workflow to receive notifications through Amazon EventBridge. See https://developer-docs.amazon.com/sp-api/docs/notifications-api-v1-use-case-guide#notification-workflows to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="notifications"
            tabindex="0"
            type="button"
          >
            Notifications
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          aria-label="Workflow to update an existing listings item. Refer to https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#updating-a-listings-item to learn more"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17ofi9h-MuiButtonBase-root-MuiButton-root"
            data-testid="updateListing"
            tabindex="0"
            type="button"
          >
            Update Listing
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
